File: src/main/java/org/example/kalkulationsprogramm/domain/EmailBlacklistEntry.java
package org.example.kalkulationsprogramm.domain;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "email_blacklist_entry", indexes = {
    @Index(name = "idx_blacklist_email", columnList = "emailAddress", unique = true)
})
@Getter
@Setter
@NoArgsConstructor
public class EmailBlacklistEntry {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true)
    private String emailAddress;

    private LocalDateTime blockedAt;

    private String blockedBy;

    public EmailBlacklistEntry(String emailAddress) {
        this.emailAddress = emailAddress != null ? emailAddress.toLowerCase() : null;
        this.blockedAt = LocalDateTime.now();
    }
}

File: src/main/kotlin/org/example/kalkulationsprogramm/domain/EmailBlacklistEntry.kt
package org.example.kalkulationsprogramm.domain

import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table
open class EmailBlacklistEntry() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    open var id: Long? = null

    @Column(nullable = false, unique = true)
    open var emailAddress: String? = null

    open var blockedAt: LocalDateTime? = null

    open var blockedBy: String? = null

    constructor(emailAddress: String?) : this() {
        this.emailAddress = emailAddress?.lowercase()
        this.blockedAt = LocalDateTime.now()
    }

}